parquet-format-structures/src/main/java/org/apache/parquet/format/Util.java

```diff
@@ -155,13 +226,33 @@ public void addRowGroup(RowGroup rowGroup) {
     public void addKeyValueMetaData(KeyValue kv) {
       md.addToKey_value_metadata(kv);
     }
+    
+    @Override
+    public void setEncryptionAlgorithm(EncryptionAlgorithm encryptionAlgorithm) {
+      md.setEncryption_algorithm(encryptionAlgorithm);
+    }
+
+    @Override
+    public void setFooterSigningKeyMetadata(byte[] footerSigningKeyMetadata) {
+      md.setFooter_signing_key_metadata(footerSigningKeyMetadata);
+    }
   }
 
   public static void readFileMetaData(InputStream from, FileMetaDataConsumer consumer) throws IOException {
-    readFileMetaData(from, consumer, false);
+    readFileMetaData(from, consumer, (BlockCipher.Decryptor) null, (byte[]) null);
```

Comment:
I would omit the casting since it doesn't add any value.

Reply:
as above.